toqito/state_props/common_epistemic_overlap.py

```diff
@@ -0,0 +1,91 @@
+"""Calculate the common epistemic overlap of quantum states."""
+
+from itertools import product
+
+import numpy as np
+
+from toqito.matrix_ops.vec import vec
+
+
+def common_epistemic_overlap(states, dim=None) -> float:
+    r"""Compute the epistemic overlap :cite:`Sagnik_2024_Epistemic`.
+
+    For a set of quantum states :math:`\{\rho_i\}` , the epistemic overlap represents the common region
+    where the probability distributions of these quantum states overlap in the underlying ontic state space.
+    In Einstein's epistemic model of quantum mechanics,
+    this overlap quantifies the extent to which different quantum states
+    share the same underlying physical reality.
+    Mathematically, it is defined as:
+
+    .. math::
+        \omega_E(\rho_1,\ldots,\rho_n) = \int \min_{\lambda\in\Lambda}
+        (\mu(\lambda|\rho_1), \ldots, \mu(\lambda|\rho_n)) d\lambda
+
+    where :math:`\Lambda` is set of all ontic states and
+    :math:`\mu(\lambda|\rho)` is the epistemic state associated with
+    the quantum state :math:`\rho` in einstein's epistemic model.
+
+    This function accepts both state vectors and density matrices as input.
+
+    Examples
+    ==========
+    State vector inputs:
+
+    >>> from toqito.state_props import common_epistemic_overlap
+    >>> from toqito.states import bell
+    >>> round(common_epistemic_overlap([bell(0), bell(2)]),4)
+    np.float64(0.0)
+
+    Mixed state inputs:
+
+    >>> import numpy as np
+    >>> from toqito.state_props import common_epistemic_overlap
+    >>> round(common_epistemic_overlap([np.eye(2)/2, np.eye(2)/2]),4)
+    np.float64(1.0)
+
+    References
+    ==========
+    .. bibliography::
+        :filter: docname in docnames
+
+
+    :param states: List of quantum states (vectors or density matrices)
+    :param dim: Optional dimension specification for composite systems
+    :raises ValueError: For invalid inputs or unsupported dimensions
+    :return: Common epistemic overlap value between 0 and 1
+
+    """
+    density_matrices = []
+    for state in states:
+        if state.ndim == 1 or (state.ndim == 2 and (state.shape[0] == 1 or state.shape[1] == 1)):
+            v = vec(state)
+            density_matrices.append(np.outer(v, v.conj()))
+        else:
+            if state.shape[0] != state.shape[1]:
```

Comment:
There is already a function to check whether a matrix is square `toqito.matrix_props.is_square`.